### ISSUE TITLE
BigQuery: Unparsable section when using multiplication

### ISSUE THREAD

**Issue by eduard-grosu** [sqlfluff/sqlfluff#6015]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.


### What Happened

I tried running `sqlfluff lint` on `test.sql` which has the following SQL:
```sql
SELECT (1*1, 2) IN (STRUCT(1 AS a, 2 AS b))
```
But it results in a failure:
```
== [test.sql] FAIL                                                                                                                                                                                                                                                       
L:   1 | P:  12 |  PRS | Line 1, Position 12: Found unparsable section: ', 2'
WARNING: Parsing errors found and dialect is set to 'bigquery'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

Apparently if I remove the multiplication, it does not result in a failure and passes the checks:
```sql
SELECT (1, 2) IN (STRUCT(1 AS a, 2 AS b))
```
Output:
```
All Finished 📜 🎉!
```


### Expected Behaviour

I expect that `sqlfluff lint` does not result in any failure.

### Observed Behaviour

Command `sqlfluff lint` fails with the following error:

```
== [test.sql] FAIL                                                                                                                                                                                                                                                       
L:   1 | P:  12 |  PRS | Line 1, Position 12: Found unparsable section: ', 2'
WARNING: Parsing errors found and dialect is set to 'bigquery'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### How to reproduce

Minimal example:
```sql
SELECT (1*1, 2) IN (STRUCT(1 AS a, 2 AS b))
```

### Dialect

BigQuery

### Version

sqlfluff, version 3.1.0

### Configuration

```yaml
[sqlfluff]
dialect = bigquery
ignore = linting,templating
large_file_skip_byte_limit = 0
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by eduard-grosu** [comment]
Upon further analysis, I discovered that this issue only arises when the multiplication is enclosed in parentheses:
`SELECT 1*1, 2` works, but `SELECT (1*1, 2)` does not. However, `SELECT (1*1)` also works, so the problem occurs only when there are multiple items.
